pkg/gui/custom_commands.go

```diff
@@ -151,10 +153,63 @@ func (gui *Gui) handleCustomCommandKeybinding(customCommand config.CustomCommand
 						return gui.surfaceError(err)
 					}
 
+					return gui.createMenu(title, menuItems, createMenuOptions{showCancel: true})
+				}
+			case "menuFromCommand":
+				f = func() error {
+					// Collect cmd to run from config
+					cmdStr, err := gui.resolveTemplate(prompt.Command, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Collect Filter regexp
+					filter, err := gui.resolveTemplate(prompt.Filter, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Run and save output
+					message, err := gui.GitCommand.RunCommandWithOutput(cmdStr)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Need to make a menu out of what the cmd has displayed
+					candidates := []string{}
+					reg := regexp.MustCompile(filter)
+					for _, str := range strings.Split(string(message), "\n") {
+						cand := ""
+						if str == "" {
+							continue
+						}
+						for i := 1; i < (reg.NumSubexp() + 1); i++ {
+							keep := reg.ReplaceAllString(str, "${"+strconv.Itoa(i)+"}")
```

Comment:
This doesn't feel right,  
If i'm correct this means every group will show up in the result right?  
So that means i can't use regex groups to make more advanced regex matches :(  
Also if i'm reading this correct does matching `FooBar` to `((Foo))` will result in FooFoo in the output?  

I'm just writing down my thoughts here but what if we would use regex named capture groups `(?P<name>...)`
We can go the simple route here and always use a group with a name like "show" but that would mean you are limited to 1 group to show on screen.
  
To solve the above we could add another property to the config named `format` or `formatter` that formats the matches by using a go template, for example:  
```
filter: '(?P<first>\w+)\w+(?P<last>\w+)'
format: '{{ .first }}-{{ .last }}'
```
So when giving it the input `foo bar baz` it will show on screen `foo-bar`  

What are your thoughts on that?

Reply:
You're right, I also thought of named regexps but didn't know how to parse tokens from `{{ .first }}-{{ .last }}`; but I think that's the most robust solution

Reply:
I just wrote an example and i think this is doable:
```go
	line := "foo bar baz"
	filter := `(?P<first>\w+)(\s+(?P<last>\w+))?`
	format := `{{ .last }}-{{ .first }}`

	b := bytes.NewBuffer(nil)
	r := regexp.MustCompile(filter)
	t := template.Must(template.New("format").Parse(format))

	// Match the line and place the matches in a map
	out := r.FindAllStringSubmatch(line, -1)
	tmplData := map[string]string{}
	if len(out) > 0 {
		for idx, groupName := range r.SubexpNames() {
			tmplData["group_"+strconv.Itoa(idx)] = out[0][idx]
			if groupName != "" {
				tmplData[groupName] = out[0][idx]
			}
		}
	}
	t.Execute(b, tmplData)
	fmt.Println(b.String())
	// prints: bar-foo

	// This way you can also just select the group by it's number
	b.Reset()
	t = template.Must(template.New("format").Parse(`{{ .group_3 }}-{{ .group_1 }}`))
	t.Execute(b, tmplData)
	fmt.Println(b.String())
	// prints: bar-foo
```

Reply:
If we're going to keep group IDs, I thought it may be also useful to keep match IDs as well. Thanks for your assistance!